src/pybind/mgr/dashboard/frontend/src/app/shared/services/prometheus-alert.service.ts

```diff
@@ -60,30 +62,35 @@ export class PrometheusAlertService {
   }
 
   private handleAlerts(alerts: AlertmanagerAlert[]) {
-    if (this.canAlertsBeNotified) {
-      this.notifyOnAlertChanges(alerts, this.alerts);
-    }
-    this.activeAlerts = _.reduce<AlertmanagerAlert, number>(
-      alerts,
-      (result, alert) => (alert.status.state === 'active' ? ++result : result),
-      0
-    );
-    this.activeCriticalAlerts = _.reduce<AlertmanagerAlert, number>(
-      alerts,
-      (result, alert) =>
-        alert.status.state === 'active' && alert.labels.severity === 'critical' ? ++result : result,
-      0
-    );
-    this.activeWarningAlerts = _.reduce<AlertmanagerAlert, number>(
-      alerts,
-      (result, alert) =>
-        alert.status.state === 'active' && alert.labels.severity === 'warning' ? ++result : result,
-      0
-    );
-    this.alerts = alerts
-      .reverse()
-      .sort((a, b) => a.labels.severity.localeCompare(b.labels.severity));
-    this.canAlertsBeNotified = true;
+    let clusterAlerts: AlertmanagerAlert[] = [];
+    this.healthService.getClusterFsid().subscribe((data: string) => {
+      const fsid = data;
```

Comment:
This is again a service, filter the alert in component and pass it here to handle it.

Reply:
Also there should filtering functionality in API itself as well.
Example: `api/v2/alerts/groups?filter=namespace=demo-dev&filter=alertname=DemoServiceRandomSimpleGauge`